backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHTransformerApi.scala

```diff
@@ -103,6 +103,22 @@ class CHTransformerApi extends TransformerApi with Logging {
   override def postProcessNativeConfig(
       nativeConfMap: util.Map[String, String],
       backendPrefix: String): Unit = {
-    /// TODO: IMPLEMENT POST PROCESS FOR CLICKHOUSE BACKEND
+    // IMPLEMENT POST PROCESS FOR CLICKHOUSE BACKEND
+    val settingPrefix = backendPrefix + ".runtime_settings."
+    if (nativeConfMap.getOrDefault("spark.memory.offHeap.enabled", "false").toBoolean) {
+      val offHeapSize =
+        nativeConfMap.getOrDefault("spark.gluten.memory.offHeap.size.in.bytes", "0").toLong
+      if (offHeapSize > 0) {
+        val maxBytesBeforeExteralSort = offHeapSize * 0.5
+        nativeConfMap.put(
+          settingPrefix + "max_bytes_before_external_sort",
+          maxBytesBeforeExteralSort.toLong.toString)
+
+        val maxBytesBeforeExternalGroupBy = offHeapSize * 0.5
```

Comment:
does it need to reserve some memory for other opertors ?

Reply:
We can't control how many memory every operator used precisely. I'm just following the suggestion of ClickHouse
```
    M(UInt64, max_bytes_before_external_group_by, 0, "If memory usage during GROUP BY operation is exceeding this threshold in bytes, activate the 'external aggregation' mode (spill data to disk). Recommended value is half of available system memory.", 0) \

    M(UInt64, max_bytes_before_external_sort, 0, "If memory usage during ORDER BY operation is exceeding this threshold in bytes, activate the 'external sorting' mode (spill data to disk). Recommended value is half of available system memory.", 0) \

```